sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/aio/_sip_routing_client_async.py

```diff
@@ -0,0 +1,267 @@
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        config = await self._rest_service.get_sip_configuration(
+            **kwargs
+        )
+        return config.routes
+
+    @distributed_trace_async
+    async def set_trunks(
+        self,
+        trunks,  # type: List[SipTrunk]
+        **kwargs  # type: Any
+    ):  # type: (...) -> None
+        """Overwrites the list of SIP trunks.
+
+        :param trunks: New list of trunks to be set.
+        :type trunks: List[SipTrunk]
+        :returns: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+        """
+        if trunks is None:
+            raise ValueError("Parameter 'trunks' must not be None.")
+
+        old_trunks = await self._get_trunks_(**kwargs)
+
+        if len(old_trunks) > 0:
+            await self._rest_service.patch_sip_configuration(
+                body=SipConfiguration(trunks={x.fqdn: None for x in old_trunks}), **kwargs
+            )
+        if len(trunks) > 0:
+            await self._patch_trunks_(trunks, **kwargs)
+
+    @distributed_trace_async
+    async def set_routes(
+        self,
+        routes,  # type: List[SipTrunkRoute]
+        **kwargs  # type: Any
+    ):  # type: (...) -> None
+        """Overwrites the list of SIP routes.
+
+        :param routes: New list of routes to be set.
+        :type routes: List[SipTrunkRoute]
+        :returns: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+        """
+        if routes is None:
+            raise ValueError("Parameter 'routes' must not be None.")
+
+        await self._rest_service.patch_sip_configuration(
+            body=SipConfiguration(routes=routes),
+            **kwargs
+            )
+
+    async def _get_trunks_(self, **kwargs):
+        config = await self._rest_service.get_sip_configuration(
+            **kwargs
+        )
+        return [SipTrunk(fqdn=k,sip_signaling_port=v.sip_signaling_port) for k,v in config.trunks.items()]
+
+    async def _patch_trunks_(self,
+        trunks,  # type: List[SipTrunk]
+        **kwargs  # type: Any
+        ):  # type: (...) -> SipTrunk
+        trunks_internal = {x.fqdn: SipTrunkInternal(sip_signaling_port=x.sip_signaling_port) for x in trunks}
+        modified_config = SipConfiguration(trunks=trunks_internal)
+
+        new_config = await self._rest_service.patch_sip_configuration(
+            body=modified_config, **kwargs
+        )
+        return [SipTrunk(fqdn=k,sip_signaling_port=v.sip_signaling_port) for k,v in new_config.trunks.items()]
+
+    async def close(self) -> None:
+        await self._rest_service.close()
+
+    async def __aenter__(self) -> "SipRoutingClient":
+        await self._rest_service.__aenter__()
+        return self
+
+    async def __aexit__(self, *args) -> None:
+        await self._rest_service.__aexit__(*args)
```

Comment:
Could we please add the `close`/`__enter__`/`__exit__` overrides to the sync client as well?